File: 14-LongestCommonPrefix/14-LongestCommonPrefix.py
# Last updated: 2/3/2026, 9:44:29 PM
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:

        strs.sort()
        first=strs[0]
        last=strs[-1]
        mini=min(len(first),len(last))
        res=""
        for i in range(mini):
            if first[i]!=last[i]:
                return res
            res+=first[i]
        return res

        

File: 14-LongestCommonPrefix/14-LongestCommonPrefix.java
// Last updated: 2/3/2026, 9:44:44 PM
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if(strs.length==0)
         return "";
         String prefix=strs[0];
         for(int i=0;i<strs.length;i++){
             while(strs[i].indexOf(prefix)!=0){
                 prefix=prefix.substring(0,prefix.length()-1);
             }
         }return prefix;
    }
}